rust-runtime/smithy-json/src/escape.rs

```diff
@@ -53,16 +62,32 @@ mod test {
             escape_string("\u{08}f\u{0C}o\to\r\n").as_ref()
         );
         assert_eq!("\\\"test\\\"", escape_string("\"test\"").as_ref());
+        assert_eq!("\\u0000", escape_string("\u{0}").as_ref());
+        assert_eq!("\\u001f", escape_string("\u{1f}").as_ref());
     }
 
     use proptest::proptest;
     proptest! {
         #[test]
-        fn matches_serde_json(s: String) {
+        fn matches_serde_json(s in ".*") {
             assert_eq!(
                 serde_json::to_string(&s).unwrap(),
                 format!(r#""{}""#, escape_string(&s))
```

Comment:
```suggestion
                escape_string(&s)
```
`Cow` and `String` are comparable